crates/sqlexec/src/planner/physical_plan/remote_scan.rs

```diff
@@ -115,7 +122,12 @@ impl ExecutionPlan for RemoteScanExec {
                 .scan(&state, projection.as_ref(), &filters, limit)
                 .await?;
 
-            plan.execute(partition, context)
+            // NOTE: RemoteScanExec can only have 1 partition since we don't
+            // know about the paritions until we "execute" the remote table
+            // which happens, unfortunately, during execution here (see above).
+            // Hence, to execute the complete plan we need to coalesce the plan.
```

Comment:
is there a way to, eventually get around this?

Reply:
With a bit of refactoring, I'm sure we could. Right now we're storing the `TableProvider` and so we don't have any info about the partitioning. Which leads to this above:

```
    fn output_partitioning(&self) -> Partitioning {
        Partitioning::UnknownPartitioning(1)
    }
```

The coalesce makes this correct.

What we can look into is store an `ExecutionPlan` which would let us get the partitioning info (and would avoid some of the above work for running the scan).

Reply:
> What we can look into is store an ExecutionPlan which would let us get the partitioning info (and would avoid some of the above work for running the scan).

This would require an additional RPC call, I was under the assumption that this was removed intentionally. I remember implementing a store for execution plans which was later removed. Is it because we were on our way to serialize all execs? (IIRC data source execs require some extra effort to serialize)

Reply:
There might've been an implementation of `StubRemoteTableProvider::scan` that did make a grpc call to create the plan, but I don't remember what it looked like. I think that is something we could add back in with relative ease. The response would just include the partitioning/execution plan id and we just wrap that in its own execution plan.